src/node/coinstats.cpp

```diff
@@ -42,8 +43,20 @@ static void ApplyHash(CCoinsStats &stats, CHashWriter& ss, const uint256& hash,
 
 static void ApplyHash(CCoinsStats& stats, std::nullptr_t, const uint256& hash, const std::map<uint32_t, Coin>& outputs, std::map<uint32_t, Coin>::const_iterator it) {}
 
+static void ApplyHash(CCoinsStats& stats, MuHash3072& muhash, const uint256& hash, const std::map<uint32_t, Coin>& outputs, std::map<uint32_t, Coin>::const_iterator it)
+{
+    COutPoint outpoint = COutPoint(hash, it->first);
+    Coin coin = it->second;
+
+    CDataStream ss(SER_DISK, PROTOCOL_VERSION);
+    ss << outpoint;
+    ss << (uint32_t)(coin.nHeight * 2 + coin.fCoinBase);
+    ss << coin.out;
+    muhash.Insert(MakeUCharSpan(ss.str()));
```

Comment:
In commit "rpc: Add hash_type MUHASH to gettxoutsetinfo"

I think extracting an `std::string` here is unnecessary. 

```c++
muhash.Insert(MakeUCharSpan(ss));
```

should work just as well, and avoid the copy.

Reply:
Fixed